templates/repo/settings/options.tmpl

```diff
@@ -966,6 +997,50 @@
 	</div>
 	{{end}}
 
+	{{if not .Repository.IsFork}}
+	<div class="ui small modal" id="repo-visibility-modal">
+		<div class="header">
+			{{ctx.Locale.Tr "repo.visibility"}}
+		</div>
+		<div class="content">
+			<div class="ui warning message">
+				{{if not .Repository.IsPrivate}}
+				{{ctx.Locale.Tr "repo.stars_remove_warning"}}<br>
+				{{ctx.Locale.Tr "repo.visibility_fork_helper"}}
+				{{else}}
+				{{ctx.Locale.Tr "repo.visibility_credential"}}
+				{{end}}
+			</div>
+			<form class="ui form" action="{{.Link}}" method="post">
+				{{.CsrfTokenHtml}}
+				<input type="hidden" name="action" value="change-visiblity">
+				<input type="hidden" name="private" value={{if .Repository.IsPrivate}}"false"{{else}}"true"{{end}}>
+				<div class="field">
+					<label>
+						{{ctx.Locale.Tr "repo.settings.transfer_form_title"}}
+						<span class="text red">{{.Repository.Name}}</span>
+					</label>
+				</div>
+				<div class="required field">
+					<label for="repo_name">{{ctx.Locale.Tr "repo.repo_name"}}</label>
+					<input id="repo_name" name="repo_name" required>
```

Comment:
There are many `id="repo_name"` elements in this page. The "id" should be unique.